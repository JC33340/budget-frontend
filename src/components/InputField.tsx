import { ChangeEvent } from 'react';

type InputFieldType = {
    placeholder: string;
    handleChange: (e: ChangeEvent<HTMLInputElement>) => void;
    value: string | number | undefined;
    className?: string;
    name: string;
    type?: string;
    maxLength?: number;
};

const InputField = ({
    placeholder,
    handleChange,
    value,
    className,
    name,
    type = '',
    maxLength,
}: InputFieldType) => {
    return (
        <div className="flex flex-col" data-testid="inputField-testid">
            <label className="text-dark-gray">{placeholder}:</label>
            <input
                className={`w-full border-1 border-dark-gray rounded-md p-2 ${className}`}
                placeholder={placeholder}
                onChange={handleChange}
                value={value}
                name={name}
                type={type}
                maxLength={maxLength}
                data-testid="inputElement-testid"
            ></input>
        </div>
    );
};

export default InputField;
