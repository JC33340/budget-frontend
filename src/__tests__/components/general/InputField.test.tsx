import InputField from '../../../components/InputField';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

test('snapshot test', () => {
    render(
        <InputField
            placeholder="test-text"
            handleChange={() => {}}
            value="test-text"
            name="test-name"
        />
    );
    const input = screen.getByTestId('inputField-testid');
    expect(input).toMatchSnapshot();
});

test('testing input function', async () => {
    const testFunc = jest.fn();
    render(
        <InputField
            placeholder="test-text"
            handleChange={testFunc}
            value="test-text"
            name="test-name"
        />
    );
    const input = screen.getByTestId('inputElement-testid');
    await userEvent.type(input, 'input');
    expect(testFunc).toHaveBeenCalledTimes(5);
});
