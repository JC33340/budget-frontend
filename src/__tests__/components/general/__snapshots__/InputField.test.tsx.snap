// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="flex flex-col"
  data-testid="inputField-testid"
>
  <label
    class="text-dark-gray"
  >
    test-text
    :
  </label>
  <input
    class="w-full border-1 border-dark-gray rounded-md p-2 undefined"
    data-testid="inputElement-testid"
    name="test-name"
    placeholder="test-text"
    type=""
    value="test-text"
  />
</div>
`;
